src/main/java/org/zalando/nakadi/service/CursorsService.java

```diff
@@ -80,11 +81,37 @@ public CursorsService(final ZooKeeperHolder zkHolder,
 
         validateStreamId(cursors, streamId, subscriptionId);
 
+        LOG.debug("[COMMIT_CURSORS] stream IDs validation finished");
+
+        final Iterator<EventTypePartition> distinctPartitions = cursors.stream()
+                .map(c -> new EventTypePartition(c.getEventType(), c.getPartition()))
+                .distinct()
```

Comment:
This looks strange that we can not provide comparator for that and instead we have a separate class to hold the values. I see that EventTypePartition class is used only in CursorsService, do you think it will good to have as inner class?